sql/lesson3/tutorial.md

```diff
@@ -0,0 +1,229 @@
+---
+layout: page
+title: Introduction to SQL
+---
+
+## About Lesson 3
+
+_Lesson 3_ will extend your knowledge of the **SELECT** operator by explaining some new functions and operators. You'll continue working with an online database already filled in with some data. At the end, you'll find some exercises which build on what you've learned in the first 3 lessons.
+
+
+### Using aggregation Functions
+When you want to start analyzing your data, it might be handy to be able to find some figures among your records. Figures like maximum, average or minimum values; or simply the total number of records in a table. To accomplish that, you can use some SQL basic Functions
+
+#### Using COUNT function
+
+**COUNT** function returns the number of records in the result. Open this lesson fiddle (https://sqliteonline.com/#fiddle-5be888fa749e451ljodase3y) and run the following query. We'll analyze the result right after.
+
+```SQL
+SELECT COUNT(*)
+  FROM people;
+```
+
+How many rows did you get in the result? Only one, right? This is because this function aggregates the whole result in one single value. In this case, the **number of records** in the **people** table.
+
+What happens if you combine this function with a WHERE statement? Something like this:
+```SQL
+SELECT COUNT(*)
+  FROM people
+ WHERE number_of_children = 2;
+```
+
+Which records is the COUNT function considering now?
+
+#### Using MIN, AVG and MAX functions
+
+Similarly, **MIN**, **AVG**, and **MAX** functions will return the minimum, average or maximum value of the given column. Contrary to COUNT, in this case the name of a column must be given as parameter. Look at these examples:
+
+```SQL
+SELECT MIN(age)
+  FROM people;
+
+SELECT MAX(age)
+  FROM people;
+```
+
+As with COUNT, you can add WHERE conditions to your statements:
+
+```SQL
+SELECT MIN(age)
+  FROM people
+ WHERE country_of_residence = 'ES';
+
+SELECT MAX(age)
+  FROM people
+ WHERE country_of_birth = 'UK';
+
+SELECT AVG(age)
+  FROM people
+ WHERE number_of_children = 1;
+```
+
+#### Using SUM function
+
+The last of the most common functions we'll explain here is **SUM**. This function adds all the values for the column given as parameter. SUM(number_of_children), for example, will sum up all the number of children on each row and reply with the total.
+
+```SQL
+SELECT SUM(number_of_children)
+  FROM people;
+```
+
+#### Combining them
+
+One way of combining them is by adding more than one function in the SELECT statement. For example:
+```SQL
+SELECT MAX(age), MIN(age),
+       MAX(number_of_children), SUM(number_of_children)
+  FROM people;
+```
+
+### Aliases
```

Comment:
Worth showing `AS` syntax too? I think it's technically more standard.

Reply:
Also, this is the first place where you start to refer to more complicated field names - worth mentioning backticks?

Reply:
Added AS as indeed is the approach in the selected style guide. Backticks are not in the style guide so I propose to leave the queries as they are now. This can also avoid confusion with the quotes around the aliases names.

Do you agree?

Reply:
Yeah, backticks and single quotes could well be synonymous in use, actually, or MySQL-specific. If normal quotes work, that's fine. I didn't notice if you explicitly mentioned them, which might be worth doing, but not essential.